src/server/gc_worker/compaction_filter.rs

```diff
@@ -232,41 +235,34 @@ impl CompactionFilter for WriteCompactionFilter {
             self.key_prefix.extend_from_slice(key_prefix);
             self.remove_older = false;
             self.switch_key_metrics();
```

Comment:
The frequency of `switch_key_metrics` is too high. We shall report the number of every  key? I do not think it is necessary.

Reply:
Just keep these 2 metris same with traditional GC. We can remove them from compaction filter and traditional GC in an another PR later.

Reply:
No. Every key will report once evenif they do not need to gc.

Reply:
I have changed it. Now it only reports when there are any versions older than safe point. It's same with `src/storage/mvcc/txn.rs:279`.

Reply:
But the judgement `if commit_ts > safe_point` is after 'switch_key_metrics'. If a key has many versions but every version is greater than safe_pointer, it will be reported in many times. I do not think it is good to report the number of version for every key. At least you should make it as a local metric.

Reply:
Em..., so the meaning of these 2 metrics are changed for compaction filter. I will fix it in an another PR.